cli/containerdutils/hostpaths.go

```
@@ -0,0 +1,294 @@
+	if err != nil {
+		return "", fmt.Errorf("failed to verify host dir %s: %s", configDir, err)
+	}
+
+	// TODO - consider refining this to be a generic utility to create a file on the host
+	// with the file content passed in as a string
+	testerSpec := specs.Spec{
+		Root: &specs.Root{
+			Path: "rootfs",
+		},
+		Process: &specs.Process{
+			Cwd: "/",
+			Args: []string{
+				"sh", "-c",
+				fmt.Sprintf("[ -f \"%s\" ] || echo \"{}\" > \"%s\" ; cat \"%s\"", configFile, configFile, configFile),
+			},
+			User: specs.User{
+				UID: 0,
+				GID: 0,
+			},
+			Env: []string{
+				"PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin",
+			},
+			NoNewPrivileges: false,
+		},
+	}
+	name := stringid.GenerateRandomID()[:20]
+	cOpts := []containerd.NewContainerOpts{
+		containerd.WithNewSnapshot(name, image),
+		containerd.WithImage(image),
+		containerd.WithSpec(&testerSpec,
+			WithHostMounts,
+			WithAllCapabilities,
+			WithHostPropagation,
+		),
+	}
+	testerSpec.Mounts = []specs.Mount{
+		{
+			Options: []string{
+				"rbind",
+				"rw",
+				"rshared",
+			},
+			Source:      path.Dir(configFile),
+			Destination: path.Dir(configFile),
+			Type:        "bind",
+		},
+	}
+	container, err := client.NewContainer(
+		ctx,
+		name,
+		cOpts...,
+	)
+	if err != nil {
+		return "", fmt.Errorf("Failed to verify config file %s: %s", configFile, err)
+	}
+	// Capture the output
+	buf := new(bytes.Buffer)
+	task, err := container.NewTask(ctx, defaultNewCreator(withByteBuffers(buf)))
+	if err != nil {
+		return "", fmt.Errorf("Failed to verify %s: %s", configFile, err)
+	}
+	err = task.Start(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Failed to start container task: %s", err)
+	}
+	status, err := task.Wait(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Failed to wait for container task: %s", err)
+	}
+	<-status
+	err = task.Kill(ctx, SIGKILL)
+	if err != nil {
+		logrus.Debugf("Failed to kill task: %s", err)
```

Same question as above, why panics instead of returning an error here ?